### PR TITLE
Some parts of API reference translation

### PR DESCRIPTION
**Pull request by chenghuzi**
For this PR, you can check the translation for the following parts in API reference:

* optuna
* optuna.cli
* optuna.distributions
* optuna.exceptions

### CODE REVIEW

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.TrialPruned.po [hunk 1]
```diff
@@ -23,21 +23,22 @@ msgstr ""
...
 
 #: of optuna.TrialPruned:3
 msgid ""
 "This error tells a trainer that the current :class:`~optuna.trial.Trial` "
 "was pruned. It is supposed to be raised after "
 ":func:`optuna.trial.Trial.should_prune` as shown in the following "
 "example."
-msgstr ""
+msgstr "该错误告诉用户 当前 :class:`~optuna.trial.Trial` 已经被剪枝. 它被设定在"
+" :func:`optuna.trial.Trial.should_prune` 之后被触发, 就像下面的例子里展示的那样."
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.TrialPruned.po]: 如下例所示，它被设定在" :func:`optuna.trial.Trial.should_prune` 之后被抛出。

I don't think the order of "它被设..., 就像下面的例子里展示的那样" is nature in Chinese.

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.create_study.po [hunk 2]
```diff
@@ -49,11 +49,13 @@ msgid ""
...
 msgstr ""
 
+
 #: of optuna.study.create_study:18
 msgid ""
 "Database URL. If this argument is set to None, in-memory storage is used,"
 " and the :class:`~optuna.study.Study` will not be persistent."
-msgstr ""
+msgstr "数据库URL. 如果该参数被设置为 None, 我们将采用内存存储, 而"
+" :class:`~optuna.study.Study` 将不可被持久化."
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.create_study.po]: 将不被持久化

Is it possible to change the storage later? If so then I think the Study "可被持久化", but it is "不被持久化" now

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.create_study.po [hunk 3]
```diff
@@ -63,7 +65,10 @@ msgid ""
 "Engine`_, you can instantiate :class:`~optuna.storages.RDBStorage` with "
 "your desired options and pass it to the ``storage`` argument instead of a"
 " URL."
-msgstr ""
+msgstr "当传入一个数据库 URL 后, Optuna 将在内部使用 `SQLAlchemy`_ 来连接"
+"数据库. 更多细节请参考 `SQLAlchemy's document`_. 如果想给 `SQLAlchemy "
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.create_study.po]: 如果想更改 `SQLAlchemyEngine`_ 的默认选项

Is it more clear?

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.create_study.po [hunk 4]
```diff
@@ -101,13 +110,14 @@ msgid ""
 "Direction of optimization. Set ``minimize`` for minimization and "
 "``maximize`` for maximization. You can also pass the corresponding "
 ":class:`~optuna.study.StudyDirection` object."
-msgstr ""
+msgstr "优化的方向. 最小化时设置为 ``minimize`` , 最大化时设置为 ``maximize``. "
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.create_study.po:113]: is the line 99-106 duplicated?

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.distributions.DiscreteUniformDistribution.po [hunk 5]
```diff
@@ -40,30 +41,34 @@ msgid ""
...
 msgid "引发"
 msgstr ""
 
 #: of optuna.distributions.DiscreteUniformDistribution:24
 msgid "If ``low`` value is larger than ``high`` value."
-msgstr ""
+msgstr "如果 ``low`` 的值比 ``high`` 大的话."
 
 #: ../../source/reference/generated/optuna.distributions.DiscreteUniformDistribution.rst:12
 msgid "Methods"
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.distributions.DiscreteUniformDistribution.po:74]: Missing translation

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.distributions.IntLogUniformDistribution.po [hunk 6]
```diff
@@ -23,44 +23,47 @@ msgstr ""
...
 
 #: of optuna.distributions.IntLogUniformDistribution:23
 msgid ""
 "Samplers and other components in Optuna relying on this distribution will"
 " ignore this value and assume that ``step`` is always 1. User-defined "
 "samplers may continue to use other values besides 1 during the "
 "deprecation."
-msgstr ""
+msgstr "Optuna 中的 sampler 和其他依赖这个分布的组件将会忽略这个值并假定 ``step`` 永远为1. "
+"在被废弃之前, 用户定义的 sampler 可能包含 1 之外的其他值."
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.distributions.IntLogUniformDistribution.po]: 用户定义的 sampler可使用1以外的其他值

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.exceptions.CLIUsageError.po [hunk 7]
```diff
@@ -23,9 +23,9 @@ msgstr ""
 
 #: of optuna.exceptions.CLIUsageError:1
 msgid "Exception for CLI."
-msgstr ""
+msgstr "命令行界面异常."
 
 #: of optuna.exceptions.CLIUsageError:3
 msgid "CLI raises this exception when it receives invalid configuration."
-msgstr ""
+msgstr "命令行界面异常在收到无效配置时会被触发."
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.exceptions.CLIUsageError.po]: 命令行界面将在收到无效的配置时抛出异常

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.exceptions.TrialPruned.po [hunk 8]
```diff
@@ -32,12 +32,18 @@ msgid ""
 ":func:`optuna.trial.Trial.should_prune` as shown in the following "
 "example."
 msgstr ""
+"该错误表明当前 :class:`~optuna.trial.Trial` "
+"已被剪枝. 它一般会在 "
+":func:`optuna.trial.Trial.should_prune` 之后被抛出, 就像下面例子中那样."
```
Comment (belldandyxtq) [inline source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.exceptions.TrialPruned.po]: ditto

**Review by belldandyxtq — CHANGES_REQUESTED** [review]

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.TrialPruned.po (hunk 1 above)
Comment (chenghuzi) [reply source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.TrialPruned.po]: Alright, I can change the order of the adverbial

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.create_study.po (hunk 2 above)
Comment (chenghuzi) [reply source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.create_study.po]: I don't think we can dump the in-memory storage to external storage. But for the sake of accuracy, will change it to "将不会被持久化", "会" is not that stronger compared with "可"

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.create_study.po (hunk 4 above)
Comment (chenghuzi) [reply source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.create_study.po:113]: Yes, that's generated by Sphinx-intl.

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.distributions.DiscreteUniformDistribution.po (hunk 5 above)
Comment (chenghuzi) [reply source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.distributions.DiscreteUniformDistribution.po:74]: You mean "Methods"?

Comment (belldandyxtq) [reply source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.distributions.DiscreteUniformDistribution.po:74]: I meant the `If ``low`` value is larger than ``high`` value.`

**Review by belldandyxtq — APPROVED** [review]

File: source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.distributions.DiscreteUniformDistribution.po (hunk 5 above)
Comment (belldandyxtq) [reply source/locale/zh_CN/LC_MESSAGES/reference/generated/optuna.distributions.DiscreteUniformDistribution.po:74]: I cannot comment on a not-changed-line

**Pull request merged**